code/datums/storage/storage.dm

```diff
@@ -1117,6 +1091,116 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 
 	modeswitch_action = null
 
+
+/**
+ * Generates UI elements for a certain UI style
+ *
+ * @param ui_style UI style that we are generating elements for
+ */
+/datum/storage/proc/generate_storage_ui(ui_style)
+	var/atom/movable/screen/close/closer = new(null, null, src)
+	closer.icon = ui_style
+	storage_closers[ui_style] = closer
+
+	var/atom/movable/screen/storage/cells = new(null, null, src)
+	var/atom/movable/screen/storage/corner/corner1 = new(null, null, src)
+	var/atom/movable/screen/storage/corner/corner2 = new(null, null, src)
+	var/atom/movable/screen/storage/corner/corner3 = new(null, null, src)
+	var/atom/movable/screen/storage/corner/corner4 = new(null, null, src)
+	var/atom/movable/screen/storage/rowjoin/rowjoin1 = new(null, null, src)
+	var/atom/movable/screen/storage/rowjoin/rowjoin2 = new(null, null, src)
+
+	cells.icon = ui_style
+	corner1.icon = ui_style
+	corner2.icon = ui_style
+	corner3.icon = ui_style
+	corner4.icon = ui_style
+	rowjoin1.icon = ui_style
+	rowjoin2.icon = ui_style
+
+	corner2.icon_state = "storage_corner_topright"
+	corner3.icon_state = "storage_corner_bottomleft"
+	corner4.icon_state = "storage_corner_bottomright"
+	corner2.update_appearance()
+	corner3.update_appearance()
+	corner4.update_appearance()
+
+	rowjoin2.icon_state = "storage_rowjoin_right"
+
+	storage_sets[ui_style] = list(cells, corner1, corner2, corner3, corner4, rowjoin1, rowjoin2)
```

Comment:
this is like really bad form. can you explain its purpose and also consider making it a datum?